components/core/src/clp_s/ArchiveWriter.cpp

```diff
@@ -67,6 +92,113 @@ void ArchiveWriter::close() {
     m_compressed_size = 0UL;
 }
 
+void ArchiveWriter::write_single_file_archive(
+        std::vector<ArchiveFileInfo> const& files,
+        size_t timestamp_dict_compressed_size
+) {
+    std::string archive_path = m_archive_path + constants::cArchiveFile;
+    FileWriter archive_writer;
+    archive_writer.open(archive_path, FileWriter::OpenMode::CreateForWriting);
+
+    write_archive_metadata(archive_writer, files, timestamp_dict_compressed_size);
+    size_t metadata_section_size = archive_writer.get_pos() - sizeof(ArchiveHeader);
+    write_archive_files(archive_writer, files);
+    m_compressed_size = archive_writer.get_pos();
+    write_archive_header(archive_writer, metadata_section_size);
+
+    archive_writer.close();
+}
+
+void ArchiveWriter::write_archive_metadata(
+        FileWriter& archive_writer,
+        std::vector<ArchiveFileInfo> const& files,
+        size_t timestamp_dict_compressed_size
+) {
+    archive_writer.seek_from_begin(sizeof(ArchiveHeader));
+
+    ZstdCompressor compressor;
+    compressor.open(archive_writer, m_compression_level);
+    compressor.write_numeric_value(3);  // Number of packets
+
+    // Write archive info
+    ArchiveInfoPacket archive_info{.num_segments = 1};
+    std::stringstream msgpack_buffer;
+    msgpack::pack(msgpack_buffer, archive_info);
+    std::string archive_info_str = msgpack_buffer.str();
+    compressor.write_numeric_value(ArchiveMetadataPacketType::ArchiveInfo);
+    compressor.write_numeric_value(archive_info_str.size());
+    compressor.write_string(archive_info_str);
+
+    // Write archive file info
+    ArchiveFileInfoPacket archive_file_info{.files{files}};
+    msgpack_buffer.clear();
```

Comment:
The clear() function on std::stringstream only clears error flags, it does not clear the stream. This means that this code just appends the new msgpack object to the previous msgpack object, creating an invalid stream of msgpack bytes.

To actually reset it we need to do something like `msgpack_buffer.str("")` or `msgpack_buffer = std::stringstream{}`.